src/components/PokeCard.js

```diff
@@ -0,0 +1,189 @@
+      transition: "transform 0.5s ease, box-shadow 0.5s ease",
+      boxShadow: "0 10px 5px rgba(0, 0, 0, 0.2)"
+    };
+  }
+
+  function styleNormalSprite() {
+    return {
+      transform: "translateZ(0px) rotateZ(0deg)"
+    };
+  }
+
+  function styleNormalText() {
+    return { transform: "translateZ(0px)" };
+  }
+
+  function stylePerspectiveCard(xAxis, yAxis) {
+    return {
+      transform: `rotateY(${xAxis}deg) rotateX(${yAxis}deg) scale(1.1)`,
+      transition: "box-shadow 0.5s ease",
+      boxShadow: "0 20px 20px rgba(0, 0, 0, 0.2)"
+    };
+  }
+
+  function stylePerspectiveSprite(xAxis, yAxis) {
+    return {
+      transform: "translateZ(130px) rotateZ(-1deg) scale(1.09)",
+      filter: `drop-shadow(${-xAxis * 1.5}px ${
+        yAxis * 1.5
+      }px 5px rgba(0, 0, 0, 0.6))`
+    };
+  }
+
+  function stylePerspectiveText(xAxis, yAxis) {
+    return {
+      transform: "translateZ(40px)",
+      filter: `drop-shadow(${-xAxis / 2}px ${
+        yAxis / 2
+      }px 1px rgba(0, 0, 0, 0.1))`
+    };
+  }
+
+  return (
+    <div style={styles.container}>
+      <Card
+        parentRef={cardRef}
+        dynamicStyle={cardStyle}
+        onMouseMove={onMoveCard}
+        onMouseLeave={onLeaveCard}
+      >
+        <CardTitle dynamicStyle={textStyle}>{mockPokemonData.name}</CardTitle>
+        <CardSprites
+          dynamicStyle={spriteStyle}
+          sprites={[
+            {
+              img: mockPokemonData.sprites.front_default,
+              alt: mockPokemonData.name
+            },
+            {
+              img: mockPokemonData.sprites.front_shiny,
+              alt: mockPokemonData.name
+            }
+          ]}
+        />
+        <CardLinkVideo href={mockPokemonData.video} dynamicStyle={textStyle}>
+          {mockPokemonData.name} video
+        </CardLinkVideo>
+      </Card>
+    </div>
+  );
+}
+
+const Card = (props) => {
+  return (
+    <div
+      ref={props.parentRef}
+      style={{ ...styles.card.container, ...props.dynamicStyle }}
+      onMouseMove={props.onMouseMove}
+      onMouseLeave={props.onMouseLeave}
+    >
+      {props.children}
+    </div>
+  );
+};
+
+const CardTitle = (props) => {
+  return (
+    <h1 style={{ ...styles.card.text, ...props.dynamicStyle }}>
+      {props.children}
+    </h1>
+  );
+};
+
+const CardSprites = (props) => {
+  return (
+    <div>
+      {props.sprites.map((sprite, i) => {
+        return (
+          <img
+            key={i}
+            src={sprite.img}
+            alt={props.alt}
+            style={{ ...styles.card.img, ...props.dynamicStyle }}
+          />
+        );
+      })}
+    </div>
+  );
+};
+
+const CardLinkVideo = (props) => {
+  return (
+    <p style={{ ...styles.card.text, ...props.dynamicStyle }}>
+      <a
+        href={props.href}
+        target="_blank"
+        rel="noreferrer"
+        style={styles.card.text}
+      >
+        {props.children}
+      </a>
+    </p>
+  );
+};
+
+const styles = {
```

Comment:
better to handle styling in a separate file

Reply:
I remove that constant and I used a css file instead, like you suggested 😃, now it looks better.

PokeCard.js
```javascript
import React, { useState, useRef } from "react";
import "./PokeCard.css";

...

const Card = (props) => {
  return (
    <div
      ref={props.parentRef}
      className="PokeCard__card"// <--- A className here
      style={{ ...props.dynamicStyle }}
      onMouseMove={props.onMouseMove}
      onMouseLeave={props.onMouseLeave}
    >
      {props.children}
    </div>
  );
};

const CardTitle = (props) => {
  return (
    <h1 className="PokeCard__text" style={{ ...props.dynamicStyle }}>// <--- A className here
      {props.children}
    </h1>
  );
};

const CardSprites = (props) => {
  return (
    <div>
      {props.sprites.map((sprite, i) => {
        return (
          <img
            key={i}
            src={sprite.img}
            alt={props.alt}
            className="PokeCard__img"// <--- A className here
            style={{ ...props.dynamicStyle }}
          />
        );
      })}
    </div>
  );
};

const CardLinkVideo = (props) => {
  return (
    <p style={{ ...props.dynamicStyle }}>
      <a
        href={props.href}
        target="_blank"
        rel="noopener noreferrer"
        className="PokeCard__text"// <--- A className here
      >
        {props.children}
      </a>
    </p>
  );
};
```